### DISCUSSION TITLE
why first_bytes plus one in path_trie.cpp

### DISCUSSION THREAD

**Discussion by zh794390558** [coqui-ai/STT#1806] [Q&A] [answered]
https://github.com/coqui-ai/STT/blob/f2e9c85880dff94115ab510cde9ca4af7ee51c19/native_client/ctcdecode/path_trie.cpp#L129-L141

**Comment by reuben** [comment, marked as answer]
In bytes output mode, label values 0-254 correspond to byte values 1-255. Label value 255 is the CTC blank symbol. The +1 here is hardcoding that assumption into the decoder code. This should definitely have a comment explaining it, so thanks for raising.

See also https://stt.readthedocs.io/en/latest/Decoder.html#bytes-output-mode
